experimental/segmenter/src/indices.rs

```diff
@@ -73,6 +75,41 @@ impl<'a> Iterator for Utf16Indices<'a> {
     }
 }
 
+#[derive(Clone)]
+pub struct PotentiallyInvalidUtf8Indices<'a> {
```

Comment:
note: if reviewing commit-by-commit, ignore this commit, I've updated the code to use [Utf8CharIndices](https://docs.rs/utf8_iter/latest/utf8_iter/struct.Utf8CharIndices.html#)